.github/workflows/ci.yml

```diff
@@ -173,50 +172,41 @@ jobs:
           SEMGREP_EXIT_CODE=$?
           set -e
-          csvData=$(node - << EOF
-          const fs = require('fs');
-          const sarifFile = fs.readFileSync("semgrep.sarif", 'utf8');
-          const sarifData = JSON.parse(sarifFile);
-
-          const tableData = [];
-
-          for (const run of sarifData.runs) {
-            for (const result of run.results) {
-              let ruleId = result.ruleId.replaceAll(',',';');
-              ruleId = ruleId.replaceAll('\n','');
-              let message = result.message.text.replaceAll(',', ';');
-              message = message.replaceAll('\n','');
-              const severity = result.level;
-              const location = result.locations[0].physicalLocation;
-              const lineNumber = location.region.startLine;
-              const fileName = location.artifactLocation.uri;
-              // Add data to the tableData array
-              tableData.push(ruleId + "|" + message +"|" + lineNumber + "|" + fileName);
-            }
-          }                  
-          // Log the CSV data as a string
-          const csvData = tableData.join('\n');
-          console.log(csvData);
-          EOF
-          )
-
-          # Append the CSV data to the output file
-          echo 'generating report'
-          echo 'report<<EOF' >> $GITHUB_OUTPUT
-          echo "$csvData" >> $GITHUB_OUTPUT
-          echo 'EOF' >> $GITHUB_OUTPUT
-          exit $SEMGREP_EXIT_CODE
+          if [ $SEMGREP_EXIT_CODE -ne 0 ]; then
+            echo "## Semgrep SAST Scan Report" > sast_report.md
+            echo "### Below are the vulnerabilities found in the code. Please resolve these before merging the pull request: " >> sast_report.md
+            echo "| Rule Id | Severity |  Message | Line Number | File Name |" >> sast_report.md
+            echo "| --- | --- | --- | --- | --- |" >> sast_report.md
+            echo 'const fs = require("fs");
+            const sarifFile = fs.readFileSync("semgrep.sarif", "utf8");
+            const sarifData = JSON.parse(sarifFile);
+            for (const run of sarifData.runs) {
+              for (const result of run.results) {
+                let ruleId = result.ruleId.replaceAll(",",";");
+                ruleId = ruleId.replaceAll("\n","");
+                let message = result.message.text.replaceAll(",", ";");
+                message = message.replaceAll("\n","");
+                const severity = result.level;
+                const location = result.locations[0].physicalLocation;
+                const lineNumber = location.region.startLine;
+                const fileName = location.artifactLocation.uri;
+                console.log("| " + ruleId + " | "  + severity + "|" + message +" | " + lineNumber + " | " + fileName + " |");
+              }
+            }' > script.js
+            node script.js >> report.md
+            cat report.md >> sast_report.md
```

Comment:
- It will be better to name script.js after what it has.
- Why is report.md is being created while it is only used to output immediately to sast_report.md